### PR TITLE
Switch from prova to tape

### LINKED ISSUE #2313: Replace prova with another test runner
Prova was unpublished from npm just now :skull: 

Unless this is a temporary glitch, we'll need to find a long-term replacement for ~~npm~~ Prova

 - https://www.npmjs.com/package/prova
 - https://github.com/azer/prova

### PR DESCRIPTION
**Pull request by lucaswoj**
cc @mourner @jfirebaugh @ansis 

fixes #2313

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +69,7 @@
     "start": "node server.js",
     "start-docs": "npm run production && npm run docs && jekyll serve -w",
     "lint": "eslint js test bench && documentation --lint",
-    "test": "npm run lint && istanbul test --include-pid --report none --print none prova -- test/js/*/*.js -q",
+    "test": "npm run lint && istanbul test --include-pid --report none --print none tape -- test/js/*/*.js | tap-dot",
```
Comment (jfirebaugh) [inline package.json]: I've had consistently bad experiences with piping tap output to some other process. It invariably breaks crash logging, `console.log`, or something else.

Can we use tap and one of its built-in reporters instead?

**Comment by lucaswoj** [comment]
Update: I'm working on a transition to `tap` but there is some niggly manual refactoring of that needs to happen first.

**Comment by lucaswoj** [comment]
In the latest commit, I switched to tap. I'm pleasantly surprised by how modern and stable tap is. It identified a number of problems with our test suite that prova was silently ignoring.

**Comment by jfirebaugh** [comment]
:+1: on code changes.

Any idea why coverage dropped over 10%? Maybe after the prova ⇢ tap switch it's counting only unit tests, or only render tests?

**Comment by lucaswoj** [comment]
Istanbul isn't playing nicely with tap. Tap has a built-in test runner but it is way too magic for us to use. Still brainstorming a solution.

**Comment by mourner** [comment]
> Istanbul isn't playing nicely with tap.

Maybe try `nyc`? There may be some interprocess stuff going on.

File: ci.sh [hunk 2]
```diff
@@ -3,35 +3,40 @@
...
 ulimit -c unlimited -S
 echo 'ulimit -c: '`ulimit -c`
 echo '/proc/sys/kernel/core_pattern: '`cat /proc/sys/kernel/core_pattern`
 sysctl kernel.core_pattern
 
-npm run test-suite
-RESULT=$?
+# run render tests
+istanbul cover --dir .nyc_output --include-pid --report none --print none test/render.test.js &&
+istanbul cover --dir .nyc_output --include-pid --report none --print none test/query.test.js
```
Comment (mourner) [inline ci.sh]: can be replaced with nyc?

Comment (lucaswoj) [inline ci.sh]: `nyc` is a wrapper around `istanbul`

The `nyc` CLI doesn't support our use case of running multiple test suites and aggregating their results. We'd have to do more file-copying or folder-renaming gymnastics to make it work.

**Pull request merged**